usr/src/boot/sys/boot/forth/menu-commands.4th

```diff
@@ -138,6 +138,26 @@ create chaincmd 1030 chars allot
 	0 boot ( state -- )
 ;
 
+\
+\ Platform-image selection for standalone SmartOS is mostly in pi.rc.
+\ We steal a boot environment routine too in pi_draw_screen (way below).
+  
+
+: init_pi ( -- )
+	s" bootpi" getenv? if else
```

Comment:
getenv? 0= if  would let to drop 'else'.

Reply:
no space after getenv, that is, we have two words, getenv will return ( len addr | -1 ), getenv? will return ( -1 | 0), 0 is false, -1 is true. getenv? is defined in support.4th.

Reply:
Dammit I typed that, I know `getenv?` is a single word.  Uggh.  Will fix.

Reply:
Also there are other "if else" antipatterns in this introduced by illumos bugfix 12448.  A cleanup for later?

Reply:
hm, missed those on review. Yea, should clean those.